src/pad/src/RDLRouter.cpp

```diff
@@ -190,113 +322,439 @@ RDLRouter::~RDLRouter()
   makeGraph();
 
   if (gui_ != nullptr) {
-    gui_->pause();
+    gui_->pause(false);
   }
 
-  std::map<odb::dbNet*, std::vector<TargetPair*>> failed;
+  // Build list of routes
+  buildIntialRouteSet();
 
-  struct RouteSet
-  {
-    TargetPair* points;
-    odb::dbNet* net;
-  };
-  std::vector<RouteSet> ordered_nets;
-  for (auto& [net, points] : routing_terminals_) {
-    for (auto& pointset : points) {
-      ordered_nets.push_back({&pointset, net});
-    }
-  }
-  std::sort(
-      ordered_nets.begin(),
-      ordered_nets.end(),
-      [](const RouteSet& r, const RouteSet& l) -> bool {
-        return distance(r.points->target0.center, r.points->target1.center)
-               < distance(l.points->target0.center, l.points->target1.center);
-      });
+  // create priority queue
+  auto route_compare
+      = [](const std::shared_ptr<RDLRoute>& lhs,
+           const std::shared_ptr<RDLRoute>& rhs) { return lhs->compare(rhs); };
+  std::priority_queue route_queue(
+      routes_.begin(), routes_.end(), route_compare);
 
   logger_->info(utl::PAD, 5, "Routing {} nets", nets.size());
-  debugPrint(logger_,
-             utl::PAD,
-             "Router",
-             1,
-             "  with {} segments",
-             ordered_nets.size());
 
-  const double dbus = block_->getDbUnitsPerMicron();
-  for (const auto& [points, net] : ordered_nets) {
-    debugPrint(logger_,
-               utl::PAD,
-               "Router",
-               2,
-               "Routing {} ({:.3f}um, {:.3f}um) -> ({:.3f}um, {:.3f}um)",
-               net->getName(),
-               points->target0.center.x() / dbus,
-               points->target0.center.y() / dbus,
-               points->target1.center.x() / dbus,
-               points->target1.center.y() / dbus);
-
-    const auto added_edges0
-        = insertTerminalVertex(points->target0, points->target1);
-    const auto added_edges1
-        = insertTerminalVertex(points->target1, points->target0);
-
-    auto route = run(points->target0.center, points->target1.center);
-    if (!route.empty()) {
+  // track destinations routed to, so we don't attempt to route to the same
+  // iterm more than once.
+  std::set<odb::dbITerm*> dst_items_routed;
+  // track sets of routes, so we don't route the reverse by accident
+  std::map<odb::dbITerm*, odb::dbITerm*> routed_pairs;
+  // track cover instances we dont route the same one twice
+  std::set<odb::dbInst*> routed_covers;
+  // track iteration information
+  int iteration_count = 0;
+  std::set<odb::dbInst*> last_itr_routed;
+  while (!route_queue.empty()) {
+    RDLRoutePtr route = route_queue.top();
+    odb::dbITerm* src = route->getTerminal();
+    route_queue.pop();
+
+    route->markRouting();
+
+    odb::dbNet* net = src->getNet();
+    auto& net_targets = routing_targets_[net];
+
+    if (routed_covers.find(src->getInst()) != routed_covers.end()) {
+      // we've already routed this cover once (indicates the cover has multiple
+      // iterms)
```

Comment:
The subsequent code is a bit unclear as to the intention.  Is this skipped or delayed or ??

Reply:
skipped, updated the comment to clarify